test/analysis-categorise.test.js

```diff
@@ -175,3 +175,12 @@ test('analysis - categorise node properties', (t) => {
 
   t.end()
 })
+
+test('do not crash on RegExp nodes', (t) => {
+  const regexpNode = new FrameNode({
+    name: '\\.js(\\?[^.]+)?$ [CODE:RegExp]'
+  })
+  regexpNode.categorise(windows)
+  t.equal(regexpNode.type, 'regexp')
```

Comment:
@AlanSl do you think we should add more assertions here?

Reply:
It'd be good to have one that ensures that a regex containing `/node_modules/` or `\\node_modules\\` doesn't get incorrectly classified as deps.

Besides that, I'd prefer to check all of this logic for edge cases but as a separate piece of work, so we can release this fix quickly.